core/src/main/scala/com/ubirch/user/core/manager/ContextManager.scala

```diff
@@ -47,26 +47,27 @@ object ContextManager extends StrictLogging
         logger.error(s"unable to update if no Context exists: contextId=$contextId")
         Future(None)
 
-      case Some(_: Context) =>
 
+      case Some(_: Context) =>
         val selector = document("id" -> contextId)
-        val update = contextWriter.write(context)
+        val update = contextHandler.writeTry(context) match {
+          case Success(doc) => doc
+          case Failure(ex) => handleError(s"parse context $context to BSONDocument", ex)
+        }
 
-        mongo.collection(collectionName) flatMap {
+        mongo.collection(collectionName).flatMap {
 
           _.update(ordered = false).one(selector, update) map { writeResult =>
-
-            if (writeResult.ok) {
+            if (writeResult.n == 1) {
               logger.info(s"updated context: id=$contextId")
               Some(context)
             } else {
-              logger.error(s"failed to update context: context=$context, writeResult=$writeResult")
-              None
+              handleError(s"update context: context=$context, writeResult=$writeResult")
```

Comment:
I'm bit confused 
when calling the function, some places are `failed to xxx` but, others are not.

Reply:
thanks! unified the remaining one.